### PR TITLE
refactor: List full token

### PR DESCRIPTION
**Pull request by MadCcc**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [x] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  -         |
| 🇨🇳 Chinese |   -        |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/_util/theme/themes/default.ts [hunk 1]
```diff
@@ -173,6 +173,7 @@ const seedToken: SeedToken = {
 
   // Line
   lineWidth: 1,
+  lineType: 'solid',
```
Comment (afc163) [inline components/_util/theme/themes/default.ts:176]: 这个和 controlLineWidth controlLineTypeidth 的区别是？

Comment (zombieJ) [reply components/_util/theme/themes/default.ts:176]: `controlLineType` 是 `lineType` 的 alias，`controlLineType` 表示为输入组件 line type。而 `lineType` 则为顶层定义。包含 Card、List 等非输入型组件。

File: components/list/style/index.tsx [hunk 2]
```diff
@@ -1,28 +1,18 @@
 // deps-lint-skip-all
 import type { CSSObject } from '@ant-design/cssinjs';
-import { TinyColor } from '@ctrl/tinycolor';
-import type { GenerateStyle, FullToken, DerivativeToken } from '../../_util/theme';
+import type { GenerateStyle, FullToken } from '../../_util/theme';
 import { resetComponent, genComponentStyleHook, mergeToken } from '../../_util/theme';
 
+export interface ComponentToken {
+  listMinWidth: number;
```
Comment (zombieJ) [inline components/list/style/index.tsx:7]: 参考 https://github.com/ant-design/ant-design/blob/next/components/cascader/style/index.tsx#L17，定 width 不用特地加 min

File: components/list/style/index.tsx [hunk 3]
```diff
@@ -372,33 +335,20 @@ const genBaseStyle: GenerateStyle<ListToken> = token => {
...
-  return [genBaseStyle(listToken), genBorderedStyle(listToken), genResponsiveStyle(listToken)];
-});
+export default genComponentStyleHook(
+  'List',
+  token => {
+    const listToken = mergeToken<ListToken>(token, {
+      listBorderedCls: `${token.componentCls}-bordered`,
+      minHeight: token.controlHeightLG,
+      listItemPadding: `${token.paddingSM}px 0`,
+      listItemPaddingSm: `${token.paddingXS}px ${token.padding}px`,
```
Comment (zombieJ) [inline components/list/style/index.tsx:345]: 是不是都应该大写大写？ SM LG

File: components/_util/theme/themes/default.ts (hunk 1 above)
Comment (afc163) [reply components/_util/theme/themes/default.ts:176]: 输入组件怎么界定？Transfer 和 Button 算输入组件么？

**Pull request merged**